0038-image-signature-verification/README.md

```diff
@@ -0,0 +1,287 @@
+- Automatic signature verification for common registries/configurations
+
+## Proposal
+
+This proposal introduces a new configuration option for Zarf package creation that allows package creators to specify Cosign verification options for container images based on registry glob pattern matching to apply specific verification rules for different registries.
+
+When a container image is pulled from a registry that matches a configured pattern, Zarf would verify its signature using the specified verification options before including it in the final package.  These options would be mapped to Cosign's CLI options to provide a familiar interface to users and allow for the same verification options to be used in both Zarf as in Cosign.
+
+### User Stories (Optional)
+
+#### Story 1: Platform Engineer Using Iron Bank Images
+
+As a platform engineer I want to validate container image signatures on package creation so that I can ensure the integrity of the images before they leave for my air-gapped environment.
+
+**Given** I have a Zarf Package images I would like to verify
+```yaml
+# zarf.yaml
+components:
+  - name: my-component
+    required: true
+    images:
+      - registry1.dso.mil/ironbank/opensource/zarf-dev/zarf/zarf-agent:v0.58.0
+      - ghcr.io/stefanprodan/podinfo:6.5.0
+```
+**When** I create that package with a `zarf-config.yaml` like the below:
+```yaml
+# zarf-config.yaml
+package:
+  create:
+    cosignOpts:
+      "registry1.dso.mil/ironbank/*":
+        certificate-chain: cosign-ca-bundle.pem
+        cert: cosign-certificate.pem
+      "ghcr.io/stefanprodan/*":
+        certificate-identity-regexp: "^https://github.com/stefanprodan/podinfo.*$"
+        certificate-oidc-issuer: https://token.actions.githubusercontent.com
+```
+**Then** Zarf will use cosign to validate the most specific glob pattern match during image pulls
+**And** Zarf will fail the package creation if the image signature verification fails
+
+### Risks and Mitigations
+
+**Performance Impact**: Signature verification adds an additional step to the image pull process, which could impact performance during package creation and deployment. To mitigate this, verification is opt-in and will be performed concurrently where possible.
+
+**UX Complexity**: Because of the number of options available in Cosign, the configuration options could become complex and may require additional documentation to help users understand the available options.  Error messages on failures will also need to be clear and actionable calling out the specific image that failed verification along with the specific reason for the failure from cosign.
+
+## Design Details
+
+### Configuration Structure
+
+The image signature verification feature will be implemented by adding a new `cosignOpts` field to the `package.create` section of the Zarf package configuration. This field will contain a map of glob patterns to verification options.
```

Comment:
Would we accept every method to verify an image that's under `cosign verify -h` as an option?

Reply:
yes I think so - there are a decent number of them but it would liekly be clearer/cleaner to have a 1-1 mapping (not sure what the library code looks like for that but hopefully there would be a way to just pull those opts straight from cosign)